## Discussion




Short read assembly failed for this genome because of the extreme repeat content.
The final draft assembly had a repeat content of **67.8%** (Table 1), with a maximum repeat size of 17.7 kb and a repeat *N*~50~ length of 485 bp.
The non-repetitive regions had an *N*~50~ length of 1066 bp.
The heterozygosity in weevil populations and lack of an inbred, laboratory strain made pooling individuals for sequencing undesirable.
Our assembly strategy of contig construction with the longest reads, followed by assembly polishing with long reads from a single individual, and then redundant contig removal with PCR-free short reads from another single individual allowed us to improve the contiguity and completeness of the stem weevil genome (Table 1).
Our final genome is draft quality and we expect gaps in the assembly at larger repeat regions that were not sufficiently covered by long reads.

We were unable to estimate historical demographics because the GBS markers were too sparse in the genome to detect runs of homozygosity.
Whole-genome resequencing, which is now widely available at low cost and high throughput, would enable these analyses.
