## Discussion

The purpose of this work was to investigate the relationship between genetic variation and resistance to biocontrol in New Zealand populations of Argentine stem weevil.
Previous reports using randomly amplified polymorphic DNA (RAPD) markers and *cytochrome C oxidase subunit I* (*COI*) sequencing suggested a high degree of genetic similarity and identified a single *COI* haplotype in New Zealand populations [@williamsGeographicalOriginIntroduced1994; @vinkPCRGutAnalysis2013].
In contrast, our results from a genome-wide genotype-by-sequencing (GBS) approach reveal a high level of genetic diversity within and between populations.
We suggest that this standing variation provides an evolutionary advantage to ASW populations in comparison to their biocontrol agent, which we expect to be exacerbated by asexual reproduction in *M. hyperodae*.
This indicates that genetic variation in both host and target need to be monitored with high-resolution genotyping to maintain success of biological control.

**Para on demographics goes here (or not)**.

Despite the increased resolution of GBS compared to traditional markers, we did not detect regions of the genome associated with parasitism by *M. hyperodae*.
Resistance to biocontrol may not be genetic, or manual detection of the parasitoid may not be a strong enough phenotype to separate resistant and susceptible individuals.
We suggest that resistance to *M. hyperodae* is encoded by multiple regions of small effect, selected from the high amount of standing variation we detected in ASW populations.
A polygenic response on standing variation is consistent with the low level of selection by the biocontrol agent [@mckenzieGeneticMolecularPhenotypic1994; @greenCisTransactingVariants2019].
The number of markers yielded by legacy GBS approaches provide low power to detect polygenic responses resulting from weak selection on standing variation.
Higher-resolution genome-wide association studies using genome-wide resequencing with more individuals and a stronger resistance phenotype may allow detection of variation associated with resistance of the weevils to biocontrol.

**Probably remove this para**.
Our final genome is draft quality and we expect gaps in the assembly at larger repeat regions that were not sufficiently covered by long reads.
Short-read genome assembly was not effective for this species because of the extreme repeat content.
Our final draft assembly had a repeat content of **67.8%** (Table 1), with a maximum repeat size of 17.7 kb and a repeat *N*~50~ length of 485 bp.
The non-repetitive regions had an *N*~50~ length of 1066 bp.
The heterozygosity in weevil populations and lack of an inbred, laboratory strain made pooling individuals for sequencing undesirable.
Our assembly strategy of contig construction with the longest reads, followed by assembly polishing with long reads from a single individual, and then redundant contig removal with PCR-free short reads from another single individual allowed us to improve the contiguity and completeness of the stem weevil genome (Table 1).

**Conclusion (let's see what demographics say)**
